src/index.d.ts

```diff
@@ -0,0 +1,256 @@
+// Type definitions for bonescript 0.6
+// Project: https://github.com/jadonk/bonescript
+// Definitions by: Troy W. <https://github.com/troywweber7>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import * as SerialPort from 'serialport';
+import * as VError from 'verror';
+
+// OBJECTS
+export const bone: BoneObj;
+
+// FUNCTIONS - TODO obliterate all "any" types where possible! --- TWW
+// https://github.com/jadonk/bonescript#system
+
+export function getPlatform(callback?: (error: ErrorType, platform: any) => void): any;
+
+export function getEeproms(callback?: (error: ErrorType, eeproms: any) => void): any;
+
+export function echo(data: any, callback?: (error: ErrorType, data: any) => void): any;
+
+export function readTextFile(filename: string, callback?: (error: ErrorType, data: any) => void): any;
+
+export function writeTextFile(filename: string, data: any, callback?: (error: ErrorType, data: any) => void): void;
+
+export function writeCModule(filename: string, data: any, callback?: (error: ErrorType, data: any) => void): void;
+
+export function setDate(date: any, callback?: (error: ErrorType, data: any) => void): void;
+
+export function analogRead(pin: string, callback?: (err: ErrorType, value: number) => void): any;
+
+export function analogWrite(pin: string, value: number, freq?: number, callback?: ErrorCb): void;
+
+export function attachInterrupt(
+    pin: string,
+    handler: InterruptFn,
+    mode: InterruptType,
+    callback?: AttachIntCb
+): void;
+
+export function detachInterrupt(pin: string, callback?: DetachIntCb): void;
+
+export function digitalRead(pin: string, callback?: (err: ErrorType, value: number) => void): any;
+
+export function digitalWrite(pin: string, value: PinStateType, callback?: DigitalWriteCb): void;
+
+export function pinMode(
+    pin: string,
+    direction: any,
+    mux?: any,
+    pullup?: any,
+    slew?: any,
```

Comment:
string. Several of these should probably should be a string enums if typescript supports that.